### PR TITLE
add ask manual question to Tanium integration

### PR DESCRIPTION
**Pull request by shaniacht1**
fixes https://github.com/demisto/etc/issues/12310
related to https://github.com/demisto/etc/issues/12834

- [x] need to improve get-sensor command to show more relevant details
- [x] add "help" arg to the ask-manual command (waiting for Alex to approve the english)
- [x] add outputs for all new commands
- [x] improve outputs for modified commands
- [x] add outputs to script
- [x] add nice outputs + nice human readable to new rest commands

### CODE REVIEW

File: Integrations/integration-Tanium.yml [hunk 1]
```diff
@@ -296,6 +318,36 @@ script:
...
+
+        if kwargs["filters_help"] or kwargs["sensors_help"] or kwargs["options_help"]:
+            try:
+                response = handler.ask(**kwargs)
+                # should always throw an exception
+                return response
+            except Exception as ex:
+                return str(ex)
+
+        kwargs["question_filters"] = args.get('question_filters', '').split(';')
```
Comment (shaniacht1) [inline Integrations/integration-Tanium.yml]: missing question options

File: Integrations/integration-Tanium.yml [hunk 2]
```diff
@@ -189,21 +188,27 @@ script:
...
         return final_result
 
     def getSavedQuestion(handler):
         response = getObject(handler, u'saved_question')
-        parsed = response['saved_question']
+        parsed = response.get('saved_question')
         final_result = create_entry('Tanium Saved Question', parsed, {'Tanium.SavedQuestions' : parsed}, ['query_text', 'name', 'id'])
         return final_result
 
+    def getSensor(handler):
```
Comment (Bars92) [inline Integrations/integration-Tanium.yml]: image
not sure why isn't `tn-get-sensor` is not working

Comment (shaniacht1) [reply Integrations/integration-Tanium.yml]: are you on demo2? because I'm working on tanium_copy there @Bars92

Comment (Bars92) [reply Integrations/integration-Tanium.yml]: yeah, ill check locally

Comment (Bars92) [reply Integrations/integration-Tanium.yml]: works good locally !

File: Integrations/integration-Tanium.yml [hunk 3]
```diff
@@ -270,17 +305,14 @@ script:
...
         return final_result
 
-    def askQuestion(handler , question):
-        kwargs = {
-            'picker': 1,
-            'question_text': question,
-            'qtype': u'parsed',
-            'get_results' : True
-        }
+    def askQuestion(kwargs):
```
Comment (shaniacht1) [inline Integrations/integration-Tanium.yml]: pass handler to func

File: Scripts/script-TaniumFilterQuestionByIndexQueryFileDetails.yml [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+  - notregex
+- name: filter_value
+  required: true
+  description: Value to filter results by
+- name: sensors
+  description: Semicolon separated list of columns to return. E.g. "Computer Name;IP
+    Address"
+  defaultValue: Computer Name
+scripttarget: 0
+runonce: false
```
Comment (shaniacht1) [inline Scripts/script-TaniumFilterQuestionByIndexQueryFileDetails.yml]: add outputs

**Comment by Bars92** [comment]
@shaniacht1 when using `tn-deploy-package` from `tn-get-all-packages`:
image
Some packages fail to work as they do not replace the package name in the command like running `Kill Process - ||Running Processes with MD5 Hash_MD5 Hash||` is running `/bin/bash taskkill_md5.sh ||Running Processes with MD5 Hash_MD5 Hash||`, but the `||Running Processes with MD5 Hash_MD5 Hash||` should have become the md5 hash to close.

File: Integrations/integration-Tanium.yml [hunk 5]
```diff
@@ -556,40 +816,57 @@ script:
...
   - name: tn-approve-pending-action
     arguments:
-    - name: action_id
-      required: true
+    - name: saved_action_id
+      description: Saved action ID to be approved
+    - name: id
       default: true
-      description: Id of saved action to approve
+      description: Action ID to be approved
```
Comment (Bars92) [inline Integrations/integration-Tanium.yml]: this is not always to be approved..

File: Integrations/integration-Tanium.yml [hunk 6]
```diff
@@ -611,66 +888,187 @@ script:
...
       description: The package associated with the action
+      type: string
     - contextPath: Tanium.PendingActions.approved_flag
       description: Whether action was approved or not
+      type: boolean
     description: Gets all pending actions
   - name: tn-get-all-sensors
     arguments: []
     description: Gets all sensors
+  - name: tn-parse-query
```
Comment (Bars92) [inline Integrations/integration-Tanium.yml]: didnt you want to remove it?

Comment (shaniacht1) [reply Integrations/integration-Tanium.yml]: didn't we decide to leave it? :O

File: Integrations/integration-Tanium.yml (hunk 5 above)
Comment (shaniacht1) [reply Integrations/integration-Tanium.yml]: @Bars92  it's an argument of tn-approve-pending-action, it has to be "to be approved"

File: Integrations/integration-Tanium.yml (hunk 6 above)
Comment (Bars92) [reply Integrations/integration-Tanium.yml]: well its good mainly for debugging

**Pull request merged**